drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/wizard/column/plugins/BRLActionColumnPlugin.java

```diff
@@ -0,0 +1,314 @@
+                                 final AdditionalInfoPage<BRLActionColumnPlugin> additionalInfoPage,
+                                 final Event<WizardPageStatusChangeEvent> changeEvent,
+                                 final TranslationService translationService) {
+        super(changeEvent,
+              translationService);
+
+        this.ruleModellerPage = ruleModellerPage;
+        this.additionalInfoPage = additionalInfoPage;
+    }
+
+    @Override
+    public String getTitle() {
+        return translate(GuidedDecisionTableErraiConstants.BRLActionColumnPlugin_AddActionBRL);
+    }
+
+    @Override
+    public List<WizardPage> getPages() {
+        return new ArrayList<WizardPage>() {{
+            add(ruleModellerPage);
+            add(additionalInfoPage());
+        }};
+    }
+
+    @Override
+    public void init(final NewGuidedDecisionTableColumnWizard wizard) {
+        super.init(wizard);
+
+        setupEditingCol();
+        setupRuleModellerEvents();
+    }
+
+    void setupRuleModellerEvents() {
+        registration = getPresenter().getEventBus().addHandler(TemplateVariablesChangedEvent.TYPE,
+                                                               this);
+    }
+
+    void setupEditingCol() {
+        editingCol = newBRLActionColumn();
+    }
+
+    void teardownRuleModellerEvents() {
+        registration.removeHandler();
+    }
+
+    @Override
+    public void onClose() {
+        super.onClose();
+
+        teardownRuleModellerEvents();
+    }
+
+    @Override
+    public Boolean generateColumn() {
+        if (nil(editingCol().getHeader())) {
+            Window.alert(translate(GuidedDecisionTableErraiConstants.BRLActionColumnPlugin_YouMustEnterAColumnHeaderValueDescription));
+            return false;
+        }
+
+        if (!isHeaderUnique(editingCol().getHeader())) {
+            Window.alert(translate(GuidedDecisionTableErraiConstants.BRLActionColumnPlugin_ThatColumnNameIsAlreadyInUsePleasePickAnother));
+            return false;
+        }
+
+        getDefinedVariables(getRuleModel());
+
+        editingCol().setDefinition(Arrays.asList(getRuleModel().rhs));
+        presenter.appendColumn(editingCol());
+
+        return true;
+    }
+
+    private boolean isHeaderUnique(String header) {
+        for (ActionCol52 o : presenter.getModel().getActionCols()) {
+            if (o.getHeader().equals(header)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    boolean getDefinedVariables(RuleModel ruleModel) {
+        Map<InterpolationVariable, Integer> ivs = new HashMap<InterpolationVariable, Integer>();
+        RuleModelVisitor rmv = new RuleModelVisitor(ivs);
+        rmv.visit(ruleModel);
+
+        //Update column and UI
+        editingCol.setChildColumns(convertInterpolationVariables(ivs));
+
+        return ivs.size() > 0;
+    }
+
+    private List<BRLActionVariableColumn> convertInterpolationVariables(Map<InterpolationVariable, Integer> ivs) {
```

Comment:
I think that it's a good refactoring, but I'm going to leave it for second PR (the editing mode), where I'm planning to do other related refactorings. Thanks, Jozef. :-)